src/main/resources/META-INF/rewrite/java-version-6.yml

```diff
@@ -0,0 +1,40 @@
+#
+# Copyright 2023 the original author or authors.
+# <p>
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+# <p>
+# https://www.apache.org/licenses/LICENSE-2.0
+# <p>
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.UpgradeToJava6
+displayName: Migrate to Java 6
+description: >
+  This recipe will apply changes commonly needed when upgrading to Java 6. This recipe will also replace deprecated API
+  with equivalents when there is a clear migration strategy.
+tags:
+  - java6
+recipeList:
+  - org.openrewrite.java.migrate.JREWrapperInterface
+---
+type: specs.openrewrite.org/v1beta/recipe
+name: org.openrewrite.java.migrate.JREWrapperInterface
+displayName: Add missing `isWrapperFor` and `unwrap` methods.
+description: Check for new methods on Java SQL interfaces.
+recipeList:
+  - org.openrewrite.java.migrate.AddMissingMethodImplementation:
+      fullyQualifiedClassName: java.sql.Wrapper
+      methodPattern: "*..* isWrapperFor(..)"
+      methodTemplateString: "public boolean isWrapperFor(Class<?> iface) throws java.sql.SQLException { return false; }"
+  - org.openrewrite.java.migrate.AddMissingMethodImplementation:
+      fullyQualifiedClassName: java.sql.Wrapper
+      methodPattern: "*..* unwrap(..)"
+      methodTemplateString: "public <T> T unwrap(Class<T> iface) throws java.sql.SQLException { return null; }"
```

Comment:
As far as I can tell this interface was added in 1.6, as were both methods contained therein? Then it does not make sense for anyone to have a class that implements that interface, yet lacks those methods, right? The compiler would not have allowed that to start with.

Should we then even have this file at all?

Reply:
Ah I only just now see that it's an interface that was added into the hierarchy, and sub classes now each need those method implementations.

Reply:
In Java 1.6 the `java.sql.Wrapper` interface was added as a subinterface to the interfaces listed in the screenshot from my original post. The `java.sql.Wrapper` interface includes a contract for the `isWrapperFor` and `unwrap` methods, meaning an implementation is required by any class implementing `java.sql.Wrapper`. So classes previously implementing `java.sql.CallableStatement`, `java.sql.Connection`, etc are now required to provide an implementation for `isWrapperFor` and `unwrap`. This recipe provides an auto-generated method stub for those methods.